perception_launch/launch/obstacle_segmentation/ground_segmentation/ground_segmentation.launch.py

```diff
@@ -318,20 +318,32 @@ def launch_setup(context, *args, **kwargs):
 
     # set container to run all required components in the same process
     container = ComposableNodeContainer(
-        name="perception_pipeline_container",
+        name=LaunchConfiguration("container_name"),
         namespace="",
         package="rclcpp_components",
         executable=LaunchConfiguration("container_executable"),
         composable_node_descriptions=[
             crop_box_filter_component,
             ground_filter_component,
         ],
+        condition=UnlessCondition(LaunchConfiguration("use_pointcloud_container")),
         output="screen",
     )
 
+    composable_nodes_loader = LoadComposableNodes(
+        composable_node_descriptions=[
+            crop_box_filter_component,
+            ground_filter_component,
+        ],
+        target_container=LaunchConfiguration("container_name"),
+        condition=IfCondition(LaunchConfiguration("use_pointcloud_container")),
+    )
+
     additional_pipeline_loader = LoadComposableNodes(
         composable_node_descriptions=additional_pipeline_components,
-        target_container=container,
+        target_container=container
+        if UnlessCondition(LaunchConfiguration("use_pointcloud_container")).evaluate(context)
+        else LaunchConfiguration("container_name"),
```

Comment:
How about to create variable for target_container?

```
target_container = containe if UnlessCondition(LaunchConfiguration("use_pointcloud_container")).evaluate(context) else LaunchConfiguration("container_name")
```

Reply:
- if use_pointcloud_container is `true`

1. UnlessCondition(LaunchConfiguration("use_pointcloud_container")).evaluate(context) is `false`
2. `target_container` is set as `LaunchConfiguration("container_name")`

```python
    concat_data_component_loader = LoadComposableNodes(
        composable_node_descriptions=[concat_data_component],
        target_container=LaunchConfiguration("container_name"),
        condition=IfCondition(
            LaunchConfiguration(
                "use_additional_pipeline",
                default=bool(ground_segmentation_param["additional_lidars"]),
            )
        ),
    )
```

- if use_pointcloud_container is `false`

1. UnlessCondition(LaunchConfiguration("use_pointcloud_container")).evaluate(context) is `true`
2. `target_container` is set as container (not use common container)

```python
    concat_data_component_loader = LoadComposableNodes(
        composable_node_descriptions=[concat_data_component],
        target_container=container,
        condition=IfCondition(
            LaunchConfiguration(
                "use_additional_pipeline",
                default=bool(ground_segmentation_param["additional_lidars"]),
            )
        ),
    )
```

Reply:
my suggestion is here [patch.txt](https://github.com/tier4/AutowareArchitectureProposal_launcher/files/7658211/patch.txt)

```
cd autoware_launcher
git applay patch.txt
```

Reply:
sorry, It may not work. I'll fix it so please wait.

Reply:
Thanks for your great suggestion :+1: 
fix in https://github.com/tier4/AutowareArchitectureProposal_launcher/pull/147/commits/4e6f733450827efc65e9866b66fae00e30d8b750

it looks good to me.
```sh
/sensing/lidar/rear/pointcloud_preprocessor/velodyne_node_container
  1  /sensing/lidar/rear/velodyne_convert_node
  2  /sensing/lidar/rear/crop_box_filter_self
  3  /sensing/lidar/rear/crop_box_filter_mirror
  4  /sensing/lidar/rear/velodyne_interpolate_node
  5  /sensing/lidar/rear/ring_outlier_filter
/sensing/lidar/left/pointcloud_preprocessor/velodyne_node_container
  1  /sensing/lidar/left/velodyne_convert_node
  2  /sensing/lidar/left/crop_box_filter_self
  3  /sensing/lidar/left/crop_box_filter_mirror
  4  /sensing/lidar/left/velodyne_interpolate_node
  5  /sensing/lidar/left/ring_outlier_filter
/perception/object_recognition/detection/euclidean/clustering/euclidean_cluster_container
  1  /perception/object_recognition/detection/euclidean/clustering/voxel_grid_filter_satoshi_x1_extreme_1898821_7703891708462921238
  2  /perception/object_recognition/detection/euclidean/clustering/outlier_filter
  3  /perception/object_recognition/detection/euclidean/clustering/euclidean_cluster
/perception/traffic_light_recognition/traffic_light_node_container
  1  /perception/traffic_light_recognition/traffic_light_image_decompressor
  2  /perception/traffic_light_recognition/traffic_light_ssd_fine_detector
  3  /perception/traffic_light_recognition/traffic_light_classifier
  4  /perception/traffic_light_recognition/traffic_light_roi_visualizer
/pointcloud_container
  1  /sensing/lidar/concatenate_data
  2  /perception/obstacle_segmentation/ground_segmentation/occupancy_grid_map_outlier_filter
  3  /perception/obstacle_segmentation/ground_segmentation/crop_box_filter
  4  /perception/occupancy_grid_map/pointcloud_to_laserscan_node
  5  /perception/obstacle_segmentation/ground_segmentation/common_ground_filter
  6  /perception/occupancy_grid_map/occupancy_grid_map_node
/map/map_container
  1  /map/lanelet2_map_loader
  2  /map/lanelet2_map_visualization
  3  /map/pointcloud_map_loader
  4  /map/map_tf_generator
/sensing/lidar/right/pointcloud_preprocessor/velodyne_node_container
  1  /sensing/lidar/right/velodyne_convert_node
  2  /sensing/lidar/right/crop_box_filter_self
  3  /sensing/lidar/right/crop_box_filter_mirror
  4  /sensing/lidar/right/velodyne_interpolate_node
  5  /sensing/lidar/right/ring_outlier_filter
/sensing/lidar/top/pointcloud_preprocessor/velodyne_node_container
  1  /sensing/lidar/top/velodyne_convert_node
  2  /sensing/lidar/top/crop_box_filter_self
  3  /sensing/lidar/top/crop_box_filter_mirror
  4  /sensing/lidar/top/velodyne_interpolate_node
  5  /sensing/lidar/top/ring_outlier_filter

```